### ISSUE TITLE
Gargoyle is Inaccessible With Screen Readers

### ISSUE THREAD

**Issue by BlindHunter95** [garglk/garglk#257]
When using Gargoyle with any screen reader, I am unable to read the text at all. This is true on Windows, Mac and Linux.

**Comment by cspiegel** [comment]
Gargoyle's handling of text makes it hard (or probably impossible) for a normal screen reader to work, since it's not using any OS routines to display the text.  Instead, it handles text drawing manually, which doesn't allow anything to really hook in and “see” the text.

On the plus side, there are at least rudimentary hooks in the code to handle TTS.  There is Windows code that exists, although you have to make some manual changes to the build system to get it to work.  I was able to create a Windows build which successfully does TTS, although it'd require a little more work to get it into proper shape.

I have no way of doing TTS on Mac, and the Linux TTS tools are pretty poor from what I've seen, but at the very least they can be stubbed out to do nothing, so at least TTS could be supported in Windows.

I'm not sure I'll be able to get things into shape for release because I'd have to do more digging to make sure it's actually working properly, but it's possible.

**Comment by cspiegel** [comment]
I've implemented TTS for both Windows and Linux.  I can't add OS X support, so I've added a feature request ticket (#263) for it and am closing this as done.
